site/docs/v1/tech/apis/webauthn.adoc

```diff
@@ -135,4 +144,140 @@ include::docs/v1/tech/apis/_import-webauthn-request-body.adoc[]
 
-include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
\ No newline at end of file
+include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
+
+== Start a WebAuthn Passkey Registration
+
+This API is used to start a WebAuthn registration ceremony by providing some details about the current user and the new passkey. The response contains the creation options that will be passed to the authenticator on the client device, including a one-time challenge unique to the current registration ceremony.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Start a WebAuthn Registration Ceremony and Retrieve Passkey Creation Options
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/api/webauthn/register/start#
+--
+
+include::docs/v1/tech/apis/_x-fusionauth-tenant-id-header-scoped-operation.adoc[]
+
+include::docs/v1/tech/apis/_webauthn-register-start-request-body.adoc[]
+
+=== Response
+
+The response for this API contains the WebAuthn passkey creation options that are passed to the WebAuthn API on the client. A few fields on the API response are returned as a base64url-encoded string and need to be converted to an `ArrayBuffer` in JavaScript before being passed to the WebAuthn client API.
+
+include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
+
+include::docs/v1/tech/apis/_webauthn-register-start-response-body.adoc[]
+
+== Complete a WebAuthn Passkey Registration
+
+This API is used to complete a WebAuthn registration ceremony by providing the values returned from the WebAuthn client API call. The API will validate the request against configured passkey requirements for the workflow and the one-time challenge generated and returned by <<Start a WebAuthn Passkey Registration>>.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:unlock[type=fas]] Complete a WebAuthn Registration Ceremony and Return Passkey Details
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/api/webauthn/register/complete#
+--
+
+include::docs/v1/tech/apis/_x-fusionauth-tenant-id-header-scoped-operation.adoc[]
+
+include::docs/v1/tech/apis/_webauthn-register-complete-request-body.adoc[]
+
+=== Response
+
+:omit_missing_header:
+include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
+:omit_missing_header!:
+
+include::docs/v1/tech/apis/_webauthn-response-body.adoc[]
+
+== Start a WebAuthn Passkey Assertion
+
+This API is used to start a WebAuthn authentication ceremony by providing some details about the current user and the new passkey. The response contains the request options that will be passed to the authenticator on the client device, including a one-time challenge unique to the current ceremony.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Start a WebAuthn Authentication Ceremony and Retrieve Passkey Request Options
```

Comment:
Why does this require an API key?

Reply:
We followed the pattern on passwordless. The APIs to start a WebAuthn ceremony require an API key. I think partly to prevent spamming the endpoint.

The APIs to complete a WebAuthn ceremony do not require an API key because the `challenge` provides another way to validate that the request is legitimate.

Reply:
Thanks!